generators/app/templates/src/app/components/Spinner/index.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { SpinnerProps } from 'react-spinkit';
+
+import Loading from './components/loading';
+import { getDisplayName } from './utils';
+
+interface WithSpinnerProps {
+  loading: boolean;
+}
+
+interface CustomSpinner extends SpinnerProps {
+  classNameContainer?: string;
+}
+
+export function withSpinner<P>(
+  WrappedComponent: React.ComponentType<P>,
+  customSpinner: CustomSpinner = {}
+): React.FC<WithSpinnerProps & P> {
```

Comment:
Just to maintain a convention adopted by some of the most known libraries, i think we should define our hoc's like this
```js
const withSomething = (/* here the params */) => (/* WrappedComponent */) => ...
```
This way we will be using the hoc the same way as, for example
```js
connect(mapStateToProps)(Component)
// or
compose(withSomeFeature, withOtherFeature)(Component)
```
What do you think about this?

Reply:
I like!